src/base/utils/string.cpp

```diff
@@ -130,6 +130,20 @@ bool NaturalCompare::lessThan(const QString &left, const QString &right)
     return false;
 }
 
+bool Utils::String::naturalCompareCaseSensitive(const QString &left, const QString &right)
+{
+    // provide a single `NaturalCompare` instance for easy use
+    static NaturalCompare nCmp(true);  // this is thread-safe in C++11 (stated in spec 6.7.4)
+    return nCmp(left, right);
+}
+
+bool Utils::String::naturalCompareCaseInsensitive(const QString &left, const QString &right)
+{
+    // provide a single `NaturalCompare` instance for easy use
+    static NaturalCompare nCmp(false);  // this is thread-safe in C++11 (stated in spec 6.7.4)
```

Comment:
>Doesn't it seem weird that the Qt developers have marked all QCollator methods as reentrant but nothing as thread-safe. 

lol, I wasn't using the same terminology as in https://doc.qt.io/qt-5/threads-reentrancy.html

>Try to declare NaturalCompare variable as static thread_local.

Cool, this will solves all problems at once. :+1: 
Also I think just `thread_local` is enough, `static` is redundant when there is `thread_local`.